org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/FeatureCallCompiler.java

```diff
@@ -1042,6 +1047,28 @@ protected void appendInlineFeatureCall(XAbstractFeatureCall call, ITreeAppendabl
 		if (prevEnd != formatString.length()) {
 			b.append(formatString.substring(prevEnd));
 		}
+		if (inlineCallNeedsParenthesis) {
+			b.append(")");
+		}
+	}
+	
+	protected boolean inlineCallNeedsParenthesis(XAbstractFeatureCall call, String formatString) {
+		String trimmedFormatString = formatString.trim();
+		if (trimmedFormatString.startsWith("(")) {
+			return false;
+		}
+		if (call.eContainer() instanceof XMemberFeatureCall) {
+			XMemberFeatureCall mfc = (XMemberFeatureCall) call.eContainer();
```

Comment:
Should we check the eContainingFeature, too?

Reply:
can you be more precise about that?

Reply:
I was thinking is parenthesis are only necessary for the target but not for the arguments.

Reply:
no i mean x.econtainingFeature == ??????

Reply:
XbasePackage.Literals.X_MEMBER_FEATURE_CALL__TARGET

Reply:
@szarnekow updated the pr

Reply:
pr updated